pkg/util/util.go

```diff
@@ -1248,3 +1248,166 @@ func DisplayLog(followLog bool, rd io.ReadCloser, compName string) (err error) {
 	return
 
 }
+
+// CopyFileWithFs copies a single file from src to dst
+func CopyFileWithFs(src, dst string, fs filesystem.Filesystem) error {
+	var err error
+	var srcinfo os.FileInfo
+
+	srcfd, err := fs.Open(src)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := srcfd.Close()
+		if err != nil {
+			klog.V(4).Infof("err occurred while closing file: %v", err)
+		}
+	}()
+
+	dstfd, err := fs.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := dstfd.Close()
+		if err != nil {
+			klog.V(4).Infof("err occurred while closing file: %v", err)
+		}
+	}()
+
+	if _, err = io.Copy(dstfd, srcfd); err != nil {
+		return err
+	}
+	if srcinfo, err = fs.Stat(src); err != nil {
+		return err
+	}
+	return fs.Chmod(dst, srcinfo.Mode())
+}
+
+// CopyDirWithFS copies a whole directory recursively
+func CopyDirWithFS(src string, dst string, fs filesystem.Filesystem) error {
+	var err error
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = fs.Stat(src); err != nil {
+		return err
+	}
+
+	if err = fs.MkdirAll(dst, srcinfo.Mode()); err != nil {
+		return err
+	}
+
+	if fds, err = fs.ReadDir(src); err != nil {
+		return err
+	}
+	for _, fd := range fds {
+		srcfp := path.Join(src, fd.Name())
+		dstfp := path.Join(dst, fd.Name())
+
+		if fd.IsDir() {
+			if err = CopyDirWithFS(srcfp, dstfp, fs); err != nil {
+				return err
+			}
+		} else {
+			if err = CopyFileWithFs(srcfp, dstfp, fs); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// StartSignalWatcher watches for kill,interrupt etc signals and cleans up temp files and folders
+func StartSignalWatcher(handle func()) {
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	defer signal.Stop(signals)
+
+	<-signals
+	handle()
+	// exit here to stop spinners from rotating
+	os.Exit(1)
+}
+
+// CleanDir cleans the original folder during events like interrupted copy etc
+// it leaves the given files behind for later use
+func CleanDir(originalPath string, leaveBehindFiles map[string]bool, fs filesystem.Filesystem) error {
+	// Open the directory.
+	outputDirRead, err := fs.Open(originalPath)
+	if err != nil {
+		return err
+	}
+
+	// Call Readdir to get all files.
+	outputDirFiles, err := outputDirRead.Readdir(0)
+	if err != nil {
+		return err
+	}
+
+	// Loop over files.
+	for _, file := range outputDirFiles {
+		if value, ok := leaveBehindFiles[file.Name()]; ok && value {
+			continue
+		}
+		err = fs.RemoveAll(filepath.Join(originalPath, file.Name()))
+		if err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// GitSubDir handles subDir for git components
+func GitSubDir(srcPath, destinationPath, subDir string, fs filesystem.Filesystem) error {
```

Comment:
well you shouldn't be exposing functions that have `fs` in them as the filesystem argument only changes when testing. 
So you should have a `gitSubDir` that has the `fs` and then the `GitSubDir` should call `gitSubDir(...., filesystem.DefaultFs{})`

Reply:
this should happen at other places too

Reply:
Fixed